### ISSUE TITLE
Composer should not warn about TLS being disabled when enabling TLS

### ISSUE THREAD

**Issue by naderman** [composer/composer#7935]
When turning disable-tls on, there is no warning currently, but there should be one. When you run composer config to reenable TLS it shows you the warning it always prints of TLS is disabled, although you are currently reenabling it.

```
$ composer config --global disable-tls true
$ composer config --global disable-tls false
You are running Composer with SSL/TLS protection disabled.
```

**Comment by Seldaek** [comment]
This is actually really hard to fix as the warning is output while creating the Composer instance, which happens *before* the config command is executed (to spin up plugins). I don't quite see how to even fix it aside from some horrible hack looking at ARGV to see if we are about to enable/disable the option.. So I doubt it's really worth doing.

**Comment by Seldaek** [comment]
Ok "fixed" in Config command instead..  so it's a bit clearer what is going on:

![image](https://user-images.githubusercontent.com/183678/51966505-fbbc4b80-246c-11e9-92cd-2948f886d818.png)

![image](https://user-images.githubusercontent.com/183678/51966517-01199600-246d-11e9-8dbe-02b5ea0d099a.png)

**Comment by stof** [comment]
that would not solve anything about the fact that the warning would be displayed before the new setting is taken into account for the `composer config` command.

**Comment by mbrodala** [comment]
That's true but it would reduce the mental burden to invert the negated option to understand what is going on. :-)
